File: 7-kyu/numbers-with-this-digit-inside/javascript/solution.js
const numbersWithDigitInside = (x, d) => {
  let foundNums = [];
  for (let i = 1; i <= x; i++) {
    if (i.toString().includes(d.toString())) {
      foundNums.push(i);
    }
  }
  const total = foundNums.reduce((acc, nxt) => acc + nxt, 0);
  const product = foundNums.length > 0 ? foundNums.reduce((acc, nxt) => acc * nxt, 1) : 0;
  return [foundNums.length, total, product];
};

File: 7-kyu/numbers-with-this-digit-inside/swift/solution.swift
import Foundation

func numbersWithDigitInside(_ x: Int64, _ d: Int64) -> [Int64] {
    let foundNums = (1...x).filter { String($0).contains(String(d)) }
    let total = foundNums.reduce(0, +)
    let product = foundNums.count > 0 ? foundNums.reduce(1, *) : 0
    return [Int64(foundNums.count), total, product]
}
